src/Sylius/Behat/Element/Admin/Product/ProductTaxonomyFormElement.php

```diff
@@ -13,106 +13,103 @@
 
 namespace Sylius\Behat\Element\Admin\Product;
 
+use Behat\Mink\Session;
+use FriendsOfBehat\SymfonyExtension\Mink\MinkParameters;
 use Sylius\Behat\Element\Admin\Crud\FormElement as BaseFormElement;
 use Sylius\Behat\Service\AutocompleteHelper;
 use Sylius\Behat\Service\DriverHelper;
+use Sylius\Behat\Service\Helper\AutocompleteHelperInterface;
 use Sylius\Component\Taxonomy\Model\TaxonInterface;
 
 final class ProductTaxonomyFormElement extends BaseFormElement implements ProductTaxonomyFormElementInterface
 {
-    public function selectMainTaxon(TaxonInterface $taxon): void
-    {
-        $this->openTaxonBookmarks();
-
-        $mainTaxonElement = $this->getElement('main_taxon')->getParent();
-
-        AutocompleteHelper::chooseValue($this->getSession(), $mainTaxonElement, $taxon->getName());
+    public function __construct(
+        Session $session,
+        array|MinkParameters $minkParameters,
+        private readonly AutocompleteHelperInterface $autocompleteHelper,
+    ) {
+        parent::__construct($session, $minkParameters);
     }
 
-    public function hasMainTaxonWithName(string $taxonName): bool
+    public function selectMainTaxon(string $taxonName): void
     {
-        $this->openTaxonBookmarks();
-        $mainTaxonElement = $this->getElement('main_taxon')->getParent();
-
-        return $taxonName === $mainTaxonElement->find('css', '.search > .text')->getText();
+        $this->changeTab();
+
+        $this->autocompleteHelper->selectByName(
+            $this->getDriver(),
+            $this->getElement('main_taxon')->getXpath(),
+            $taxonName,
+        );
+        $this->waitForFormUpdate();
     }
 
-    public function checkProductTaxon(TaxonInterface $taxon): void
+    public function getMainTaxon(): string
     {
-        $this->openTaxonBookmarks();
+        $this->changeTab();
 
-        $productTaxonsCodes = [];
-        $productTaxonsElement = $this->getElement('product_taxons');
-        if ($productTaxonsElement->getValue() !== '') {
-            $productTaxonsCodes = explode(',', $productTaxonsElement->getValue());
-        }
-        $productTaxonsCodes[] = $taxon->getCode();
-
-        $productTaxonsElement->setValue(implode(',', $productTaxonsCodes));
+        return $this
+            ->getElement('main_taxon')
+            ->find('css', 'option:selected')
+            ->getText()
+        ;
```

Comment:
Since we can have only one main taxon selected, you can just create a new element
```php
'selected_main_taxon' => '[data-test-main-taxon] option:selected'
```
and change
```suggestion
        return $this->getElement('selected_main_taxon')->getText();
```